src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,464 @@
+          // using a standard aria-label in this case.
+          aria-label={ariaLabel}
+        >
+          {children}
+        </Listbox>
+      </DropdownContext.Provider>
+    );
+  }
+
+  const label = (labelText || ariaLabel) && (
+    <DropdownLabel className={ariaLabel && styles['sr-only']}>
+      {labelText || ariaLabel}
+    </DropdownLabel>
+  );
+
+  const trigger = buttonText && <DropdownTrigger>{buttonText}</DropdownTrigger>;
+
+  const optionsList = options && (
+    <DropdownOptions>
+      {options.map((option) => {
+        const { label, ...rest } = option;
+        return (
+          <DropdownOption value={option} {...rest} key={option.key}>
+            {label}
+          </DropdownOption>
+        );
+      })}
+    </DropdownOptions>
+  );
+
+  const childrenToUse = (
+    <>
+      {label}
+      {trigger}
+      {optionsList}
+      {children}
+    </>
+  );
+
+  return (
+    <DropdownContext.Provider value={contextValue}>
+      <Listbox {...sharedProps}>{childrenToUse}</Listbox>
+    </DropdownContext.Provider>
+  );
+}
+
+const DropdownContext = React.createContext<{
+  compact?: boolean;
+  optionsAlign?: OptionsAlignType;
+  optionsClassName?: string;
+}>({});
+
+const DropdownLabel = (props: { className?: string; children: ReactNode }) => {
+  const { children, className } = props;
+
+  return (
+    <Listbox.Label className={clsx(styles['label'], className)}>
+      {children}
+    </Listbox.Label>
+  );
+};
+
+const DropdownTrigger = function (
+  props: PropsWithRenderProp<{ disabled: boolean; open: boolean }>,
+) {
+  const { children, className, ...rest } = props;
+  const { compact } = useContext(DropdownContext);
+
+  return (
+    <Listbox.Button
+      // Render as a fragment instead of the default element. We're rendering our own element in
+      // the render prop to control styling and positiong, and we don't want to end up with
+      // duplicate buttons.
+      as={React.Fragment}
+      {...rest}
+    >
+      {typeof children === 'function' ? (
+        children
+      ) : (
+        <DropdownButton
+          className={clsx(
+            className,
+            compact && styles['dropdown-button--compact'],
+          )}
+          text={children}
+        />
+      )}
+    </Listbox.Button>
+  );
+};
+
+const DropdownOptions = function (
+  props: PropsWithRenderProp<{ open: boolean }>,
+) {
+  const { className, ...rest } = props;
+  const { compact, optionsAlign, optionsClassName } =
+    useContext(DropdownContext);
+
+  const computedClassName = clsx(
+    styles['dropdown__options'],
+    className,
+    optionsAlign === 'right' && styles['dropdown__options--align-right'],
+    optionsClassName || (!compact && styles['dropdown__options--full-width']),
+  );
+  return <Listbox.Options className={computedClassName} {...rest} />;
+};
+
+const DropdownOption = function (props: DropdownOptionProps) {
+  const { children, className, ...rest } = props;
```

Comment:
Damn so much prop deconstruction in this component huh lol
```suggestion
  const { children, className, ...other } = props;
```

Reply:
Yea...